### ISSUE TITLE
Wrong navigation when go back in resource create

### ISSUE THREAD

**Issue by bocanhcam** [laravel/nova-issues#6431]
In a resource `User`:

Steps:
1. In an any page
2. Go to `User` index page
3. Click button create, go to `User` create page
2. Submit with invalid input to get error 422
3. Click on button go back of application or browser

Expected:
- Redirect to `User` resouce index

Current:
- Go back to the page before go to the User resouce index

- Laravel Version: v10.48.12
- Nova Version: 4.34.2
- PHP Version: 8.1

[Screencast from 21-06-2024 11:14:27.webm](https://github.com/laravel/nova-issues/assets/35592955/725bd4c0-9886-4fce-97b1-069ed752bb48)

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by bocanhcam** [comment]
Sorry this is not a bug.

I created a custom tool and tried to replace the `ResourceUpdate` component.

```js
import ResourceUpdate from './pages/Update'
const pages = Nova.pages

Nova.booting((app) => {
    pages['Nova.Update'].components.ResourceUpdate = ResourceUpdate
})
```

**Comment by bocanhcam** [comment]
Here is the repository
https://github.com/bocanhcam/nova-issue-6431

**Comment by bocanhcam** [comment]
[Screencast from 19-07-2024 10:04:16.webm](https://github.com/user-attachments/assets/fd809cf2-dfc7-426d-b96a-9ac8b4aaee6d)

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
